worlds/sc2/client.py

```diff
@@ -1566,19 +1566,29 @@ def calculate_kerrigan_options(ctx: SC2Context) -> int:
     return result
 
 
-def caclulate_soa_options(ctx: SC2Context) -> int:
+def caclulate_soa_options(ctx: SC2Context, mission: SC2Mission) -> int:
+    """
+    Pack SOA options into a single integer with bitflags.
+    0b000011 = SOA presence
+    0b000100 = SOA in no-builds
+    0b011000 = Passives presence
+    0b100000 = PAssives in no-builds
+    """
     result = 0
 
     # Bits 0, 1
     # SoA Calldowns available
     soa_presence_value = 0
-    if ctx.spear_of_adun_presence == SpearOfAdunPresence.option_not_present:
-        soa_presence_value = 0
-    elif ctx.spear_of_adun_presence == SpearOfAdunPresence.option_lotv_protoss:
-        soa_presence_value = 1
-    elif ctx.spear_of_adun_presence == SpearOfAdunPresence.option_protoss:
-        soa_presence_value = 2
-    elif ctx.spear_of_adun_presence == SpearOfAdunPresence.option_everywhere:
+    if (
+        (ctx.spear_of_adun_presence == SpearOfAdunPresence.option_everywhere)
+        or (ctx.spear_of_adun_presence == SpearOfAdunPresence.option_protoss and MissionFlag.Protoss in mission.flags)
+        or (ctx.spear_of_adun_presence == SpearOfAdunPresence.option_any_race_lotv and mission.campaign == SC2Campaign.LOTV)
+        or (ctx.spear_of_adun_presence == SpearOfAdunPresence.option_lotv_protoss
+            and (MissionFlag.VanillaSoa in mission.flags  # Keeps SOA off on Growing Shadow, as that's vanilla behaviour
```

Comment:
This change would need a doc update at least (says all in LotV main campaign). As the current approach is that SoA is just introduced later in the campaign and similarly you don't ban Factory/Starport in The Outlaws, even if you can't build them in vanilla mission.

I'd prefer it keeping it simple = SoA allowed for Growing Shadow as it's a part of LotV main campaign (as the option is currently documented.

Note that for API=3 games shifting the `option_lotv_protoss` to `option_any_race_lotv` keeps the prior behavior, while keeping `option_lotv_protoss` doesn't

Reply:
BTW I'd create a method for this checking function as the check isn't that simple)